cal/api/serializers.py

```diff
@@ -59,14 +60,15 @@ def get_fields(self, *args, **kwargs):
 
     class Meta:
         model = ColorCategory
-        fields = ('color_index', 'label', 'hours', 'calendar')
+        fields = ('id', 'color_index', 'label', 'hours', 'calendar')
 
     def get_hours(self, obj):
-        calendar_ids = self.context['calendar_ids']
-        start = self.context['start']
-        end = self.context['end']
+        calendar_ids = self.context.get('calendar_ids')
+        start = self.context.get('start')
+        end = self.context.get('end')
```

Comment:
Instead of changing these to `get`, you should add the same `get_serializer_context` method from `ColorCategoryList` to the `ColorCategoryDetail` class so that these three keys will always exist in the context dictionary (it's accessed in the serializer's `get_hours` method). This is because we want our API endpoints for single categories to be able to filter by calendar ids, start, and end as well. It's my fault that I missed adding this code for `TagDetail`, so while you're at it, could you add the same code to `TagDetail` as well?